src/umpire/strategy/NumaPolicyStrategy.cpp

```diff
@@ -0,0 +1,85 @@
+//////////////////////////////////////////////////////////////////////////////
+// Copyright (c) 2018-2019, Lawrence Livermore National Security, LLC.
+// Produced at the Lawrence Livermore National Laboratory
+//
+// Created by David Beckingsale, david@llnl.gov
+// LLNL-CODE-747640
+//
+// All rights reserved.
+//
+// This file is part of Umpire.
+//
+// For details, see https://github.com/LLNL/Umpire
+// Please also see the LICENSE file for MIT license.
+//////////////////////////////////////////////////////////////////////////////
+#include "umpire/strategy/NumaPolicyStrategy.hpp"
+#include "umpire/util/Macros.hpp"
+
+#include "umpire/ResourceManager.hpp"
+
+#include <numa.h>
+#include <numaif.h>
+
+namespace umpire {
+
+namespace strategy {
+
+NumaPolicyStrategy::NumaPolicyStrategy(
+    const std::string& name,
+    int id,
+    int numa_node,
+    Allocator allocator) :
+  AllocationStrategy(name, id),
+  m_node(numa_node),
+  m_mask(numa_bitmask_alloc(numa_max_node() + 1)),
+  m_allocator(allocator.getAllocationStrategy())
```

Comment:
We might want to check that the `Platform` of the passed allocator is `Platform::cpu`, since this won't work for others.

Reply:
good catch! I added a check.